src/System.ComponentModel.Composition/src/Microsoft/Internal/Collections/CollectionServices.CollectionOfObject.cs

```diff
@@ -55,17 +63,17 @@ public void Clear()
 
             public bool Contains(object item)
             {
-                return Assumes.NotReachable<bool>();
+                throw new Exception("Code path should never be reached!");
```

Comment:
I would throw a `NotImplementedException` here and in other places that aren't reachable.